BCH26.c

```diff
@@ -0,0 +1,69 @@
+#include <stdlib.h>
+#include "BCH26.h"
+#define gx 0x05B9<<(26-11)
+unsigned int CheckMatrix[26][2] = {
+        {119, 33554432},
+        {743, 16777216},
+        {943, 8388608},
+        {779, 4194304},
+        {857, 2097152},
+        {880, 1048576},
+        {440, 524288},
+        {220, 262144},
+        {110, 131072},
+        {55, 65536},
+        {711, 32768},
+        {959, 16384},
+        {771, 8192},
+        {861, 4096},
+        {882, 2048},
+        {441, 1024},
+        {512, 512},
+        {256, 256},
+        {128, 128},
+        {64, 64},
+        {32, 32},
+        {16, 16},
+        {8, 8},
+        {4, 4},
+        {2, 2},
+        {1, 1}
+};
+
+unsigned int decode_BCH_26_16(unsigned int code, unsigned int *value){
+    // this code is from: https://blog.csdn.net/u012750235/article/details/84622161
```

Comment:
I wonder, is this code under a GPL-compatible license? The custom for free software projects is that there should be a copyright notice in each file showing who holds the copyright, accompanied by license information. It is good to be careful about these things so it does not cause a problem later on.